src/util/expr_initializer.cpp

```diff
@@ -122,7 +122,7 @@ exprt expr_initializert<nondet>::expr_initializer_rec(
     {
       // we initialize this with an empty array
 
-      array_exprt value(array_type);
+      array_exprt value({}, array_type);
       value.type().id(ID_array);
```

Comment:
I don't see why this is necessary

Reply:
We could arguably make the no-operands-constructor non-deprecated. Would that seem better?

Reply:
So I'm still in two minds about this. I see the following options:

1) De-deprecate the no-operands-constructor so that such a change isn't necessary. Reduces the amount of noise in this PR.
2) Require the above syntax. Adds some noise, but has the following advantages: a) smaller API as fewer constructors are provided and b) encouraging efficient bottom-up construction.

Comments would be appreciated.